### ISSUE TITLE
Improve error messages and failure for publish service

### ISSUE THREAD

**Issue by humphd** [mozilla/thimble.mozilla.org#1797]
We have a Heroku/Amazon S3 outage right now, and when you try to remix or open a project, you get:

```
Service Unavailable (503): Failed to load Thimble!
```

That's brutal, and makes it look like its our problem, when really it's nothing to do with us.  We should deal with the 500 case and at least let you work on a "Start from Scratch" project vs. just doing nothing.

**Comment by Pomax** [comment]
yeah, adding in handling for 5xx errors is a great idea. Are there any other errors we can think of that need dealing with in the 5xx family outside of 500 and 503?

**Comment by gideonthomas** [comment]
But it won't let you load Brackets though? Since it's also on S3?

**Comment by Pomax** [comment]
think asynchronouser: say you're working in thimble already, and S3 dies.

**Comment by humphd** [comment]
Even (and especially!) if it won't load, we should provide something useful here.  Having said that, we load Brackets from Service Worker, so repeat users would still be able to load it.

**Comment by evantsyferov** [comment]
I would like to work on this issue.

**Comment by humphd** [comment]
@evantsyferov OK, it's yours.  @gideonthomas, @cadecairos can you give a few tips on good places to focus this?  I'd basically love to not have 500s make there way to the user, and instead let some kind of error middleware or error page show something instead.

**Comment by cadecairos** [comment]
Gideon can correct me if I'm wrong, but I think the [files in here](https://github.com/mozilla/thimble.mozilla.org/tree/master/server/routes) might be a good place to have a look. in the sub folders are request handlers that hit up the publish server ([such as this one](https://github.com/mozilla/thimble.mozilla.org/blob/master/server/routes/projects/publish.js#L24-L70)) that might be improved.

There also seems to be a [generic error handler ](https://github.com/mozilla/thimble.mozilla.org/blob/master/server/lib/http-error.js) that might be a good place to look into

**Comment by evantsyferov** [comment]
I done some research on the issue and have found that almost every case for errors ends up calling: next(HttpError.format(err, req));  This calls the static format(error, request) function in this [file](https://github.com/mozilla/thimble.mozilla.org/blob/master/server/lib/http-error.js)
  I am confused on were I can get code on the next function to see what it does.

**Comment by humphd** [comment]
@cadecairos, @gideonthomas can you shed some light on @evantsyferov's question above?  Thanks for any help.

**Comment by cadecairos** [comment]
@evantsyferov It looks like we set up the error handling middleware functions [here](https://github.com/mozilla/thimble.mozilla.org/blob/master/server/index.js#L127-L128) ([ and here's how error handling middleware in express works](https://expressjs.com/en/guide/error-handling.html))

Those middleware functions are defined [here](https://github.com/mozilla/thimble.mozilla.org/blob/master/server/lib/http-error.js#L6-L43) and [here](https://github.com/mozilla/thimble.mozilla.org/blob/master/server/lib/http-error.js#L45-L49)

**Comment by evantsyferov** [comment]
Through some more research I come to conclusion that code to render error.html needs to be put in the static generic and format function in this<a href= "https://github.com/mozilla/thimble.mozilla.org/blob/master/server/lib/http-error.js" > file </a>. I am thinking to add something like this:  response.render("error.html", { reponse.statuscode}); I am not sure what to put between the {}. Is this a good idea or does someone have a better suggestion?
